src/TensorFlowNET.Core/Tensors/TensorShape.cs

```diff
@@ -57,10 +57,36 @@ public TensorShape(TensorShapeProto proto)
 
         public TensorShape(params object[] dims)
         {
-            var intdims = new int[dims.Length];
```

Comment:
Why define it as `params object[] dims` ? If we define `int None = -1`, we can keep it as `params int[] dims` or `params long[] dims`. and can utilize the implicit conversion, we defined conversion for simplify syntax, for example: `public static implicit operator TensorShape((int, int) dims) => new TensorShape(dims.Item1, dims.Item2);`

Reply:
If you add `public TensorShape(params int[] dims)` the project will not compile due to ambiguous methods.

Reply:
`public static implicit operator TensorShape((int, int) dims) => new TensorShape(dims.Item1, dims.Item2);`
I'll add support if int? and long? in the constructor so you can write:
`public static implicit operator TensorShape((int?, int?) dims) => new TensorShape(dims.Item1, dims.Item2);`